src/hotspot/cpu/riscv/stubGenerator_riscv.cpp

```diff
@@ -4277,6 +4277,129 @@ class StubGenerator: public StubCodeGenerator {
     return (address) start;
   }
 
+  // rotate vector register left with shift bits, 32-bit version
+  void rotate_left_imm(VectorRegister rv, uint32_t shift, VectorRegister tmp_vr) {
+    __ vsrl_vi(tmp_vr, rv, 32 - shift);
+    __ vsll_vi(rv, rv, shift);
+    __ vor_vv(rv, rv, tmp_vr);
+  }
+
+  void quarter_round(VectorRegister aVec, VectorRegister bVec,
```

Comment:
Please rename that to `chacha20_quarter_round` to make it clear it belongs to the chacha20 algorithm

Reply:
And should be static.